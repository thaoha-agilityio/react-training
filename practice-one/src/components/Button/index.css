.btn {
  border-radius: 5px;
  font-family: 'Inter';
  border: none;
  cursor: pointer;
}

.btn-primary {
  background: var(--color-secondary-100);
  color: var(--color-secondary);
  text-transform: uppercase;
}

.btn-secondary {
  background-color: var(--color-secondary-200);
  color: var(--color-secondary);
}

.btn-tertiary {
  background-color: var(--color-tertiary);
}

.btn-normal {
  background-color: var(--color-secondary);
  color: var(--color-primary-50);
}

.btn-small {
  padding: 2px 10px;
  font-size: var(--fs-small);
}

.btn-medium {
  padding: 7px 22px;
  font-size: var(--fs-medium);
  line-height: calc(24 / 14);
}

.btn-large {
  padding: 10px 20px;
  font-size: var(--fs-large);
}
