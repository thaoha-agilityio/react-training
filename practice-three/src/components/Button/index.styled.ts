import styled from "styled-components";

interface StyledButtonProps {
  bgColor: string;
  borderColor?: string;
  width?: number;
  height?: number;
}

export const IconButtonStyled = styled.button<StyledButtonProps>`
  ${(props) => props.width && `width:${props.width}px`};
  ${(props) => props.height && `height:${props.height}px`};
  ${(props) => `background-color:${props.bgColor}`};
  border: none;
  cursor: pointer;
`;

export const CircleButtonStyled = styled(IconButtonStyled)`
  border-radius: 50%;
  border: 1px solid;
  ${(props) => props.borderColor && `border-color:${props.borderColor}`};
`;
