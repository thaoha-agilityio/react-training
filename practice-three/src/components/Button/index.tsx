import { ReactNode } from "react";

import { CircleButtonStyled, IconButtonStyled } from "./index.styled";

interface IProps extends React.ButtonHTMLAttributes<HTMLButtonElement> {
  isCircle?: boolean;
  bgColor: string;
  borderColor?: string;
  height?: number;
  width?: number;
  icon: ReactNode;
}

const Button = ({
  width = 40,
  height = 40,
  icon,
  isCircle = false,
  bgColor,
  ...rest
}: IProps): React.ReactElement => (
  <>
    {isCircle ? (
      <CircleButtonStyled
        width={width}
        height={height}
        bgColor={bgColor}
        {...rest}
      >
        {icon}
      </CircleButtonStyled>
    ) : (
      <IconButtonStyled
        width={width}
        height={height}
        bgColor={bgColor}
        {...rest}
      >
        {icon}
      </IconButtonStyled>
    )}
  </>
);

export default Button;
