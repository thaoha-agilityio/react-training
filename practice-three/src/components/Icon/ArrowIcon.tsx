import { SVGProps } from 'react';

export const ArrowIcon = (props: SVGProps<SVGSVGElement>) => (
  <svg width={8} height={11} fill="none" xmlns="http://www.w3.org/2000/svg" {...props}>
    <path
      fillRule="evenodd"
      clipRule="evenodd"
      d="M.147 7.146a.5.5 0 0 1 .708 0L3.5 9.793l2.646-2.647a.5.5 0 0 1 .708.708l-3 3a.5.5 0 0 1-.708 0l-3-3a.5.5 0 0 1 0-.708Z"
      fill="#000000"
    />
    <path
      fillRule="evenodd"
      clipRule="evenodd"
      d="M3.5 0a.5.5 0 0 1 .5.5v9a.5.5 0 1 1-1 0v-9a.5.5 0 0 1 .5-.5Z"
      fill="#000000"
    />
  </svg>
);
