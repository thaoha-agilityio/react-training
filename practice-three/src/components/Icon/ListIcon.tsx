import { SVGProps } from "react";

export const ListIcon = (props: SVGProps<SVGSVGElement>) => (
  <svg
    width={14}
    height={10}
    fill="none"
    xmlns="http://www.w3.org/2000/svg"
    {...props}
  >
    <path
      fillRule="evenodd"
      clipRule="evenodd"
      d="M4 9a.5.5 0 0 1 .5-.5h9a.5.5 0 0 1 0 1h-9A.5.5 0 0 1 4 9Zm0-4a.5.5 0 0 1 .5-.5h9a.5.5 0 0 1 0 1h-9A.5.5 0 0 1 4 5Zm0-4a.5.5 0 0 1 .5-.5h9a.5.5 0 0 1 0 1h-9A.5.5 0 0 1 4 1ZM1 2a1 1 0 1 0 0-2 1 1 0 0 0 0 2Zm0 4a1 1 0 1 0 0-2 1 1 0 0 0 0 2Zm0 4a1 1 0 1 0 0-2 1 1 0 0 0 0 2Z"
      fill="#000"
    />
  </svg>
);
