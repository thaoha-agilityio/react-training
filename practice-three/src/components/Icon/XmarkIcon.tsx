import { SVGProps } from "react";

export const XmarkIcon = (props: SVGProps<SVGSVGElement>) => (
  <svg
    width={14}
    height={14}
    fill="none"
    xmlns="http://www.w3.org/2000/svg"
    {...props}
  >
    <path
      fillRule="evenodd"
      clipRule="evenodd"
      d="M.22.239a.75.75 0 0 1 1.06 0l5.22 5.22 5.22-5.22a.75.75 0 1 1 1.06 1.06l-5.22 5.22 5.22 5.22a.75.75 0 1 1-1.06 1.06L6.5 7.579l-5.22 5.22a.75.75 0 0 1-1.06-1.06l5.22-5.22-5.22-5.22a.75.75 0 0 1 0-1.06Z"
      fill="#58667E"
    />
  </svg>
);
