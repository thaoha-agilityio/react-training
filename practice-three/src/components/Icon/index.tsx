export * from "./ArrowIcon";

export * from "./FilterIcon";

export * from "./GridIcon";

export * from "./LightIcon";

export * from "./ListIcon";

export * from "./DarkIcon";

export * from "./SearchIcon";

export * from "./XmarkIcon";

export * from "./ArrowRightIcon";
