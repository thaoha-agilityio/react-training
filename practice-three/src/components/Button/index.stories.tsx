import { ComponentStory, ComponentMeta } from "@storybook/react";

import { ListIcon, LightIcon, GridIcon } from "@/components/Icon";
import { colors } from "@/themes";

import Button from ".";

export default {
  title: "Button",
  component: Button,
  argTypes: {
    isCircle: {
      defaultValue: false,
      control: { type: "boolean" },
    },
    icon: {
      defaultValue: <LightIcon />,
    },
    width: {
      defaultValue: 40,
    },
    height: {
      defaultValue: 40,
    },
    bgColor: {
      defaultValue: "transparent",
    },
  },
  parameters: {
    actions: {
      handles: ["click"],
    },
  },
} as ComponentMeta<typeof Button>;

const Template: ComponentStory<typeof Button> = (args) => <Button {...args} />;

export const Basic = Template.bind({});

export const CircleBtn = Template.bind({});
CircleBtn.args = {
  ...CircleBtn.args,
  isCircle: true,
  icon: <ListIcon />,
  borderColor: colors.athensGray,
};

export const CircleBtnActive = Template.bind({});
CircleBtnActive.args = {
  ...CircleBtnActive.args,
  isCircle: true,
  icon: <GridIcon />,
  bgColor: colors.linkWater,
  borderColor: colors.athensGray,
};
