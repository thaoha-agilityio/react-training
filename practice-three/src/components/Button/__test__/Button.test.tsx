import { render, screen } from "@testing-library/react";
import "@testing-library/jest-dom";

import { ListIcon } from "../../Icon";

import Button from "..";

describe("testing button component", () => {
  test("testing default button", () => {
    const tree = render(
      <Button isCircle={false} variant="primary" size="small" text="click" />
    );

    const button = screen.getByRole("button");

    expect(button).toBeTruthy();
    expect(tree).toMatchSnapshot();
  });

  test("testing icon button", () => {
    const tree = render(
      <Button
        aria-label="icon"
        isCircle={false}
        variant="primary"
        size="large"
        icon={<ListIcon />}
      />
    );
    const button = screen.getByRole("button");

    expect(button).toBeTruthy();
    expect(tree).toMatchSnapshot();
  });

  test("testing circle button", () => {
    const tree = render(
      <Button
        aria-label="icon"
        isCircle={true}
        variant="primary"
        size="large"
        icon={<ListIcon />}
      />
    );

    const button = screen.getByRole("button");

    expect(button).toBeTruthy();
    expect(tree).toMatchSnapshot();
  });
});
