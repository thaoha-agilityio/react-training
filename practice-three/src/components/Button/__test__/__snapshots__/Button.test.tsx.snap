// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testing button component testing circle button 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <button
        class="sc-bcXHqe sc-gswNZR ePFmvY jAVsth"
      >
        <svg
          fill="none"
          height="10"
          width="14"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            clip-rule="evenodd"
            d="M4 9a.5.5 0 0 1 .5-.5h9a.5.5 0 0 1 0 1h-9A.5.5 0 0 1 4 9Zm0-4a.5.5 0 0 1 .5-.5h9a.5.5 0 0 1 0 1h-9A.5.5 0 0 1 4 5Zm0-4a.5.5 0 0 1 .5-.5h9a.5.5 0 0 1 0 1h-9A.5.5 0 0 1 4 1ZM1 2a1 1 0 1 0 0-2 1 1 0 0 0 0 2Zm0 4a1 1 0 1 0 0-2 1 1 0 0 0 0 2Zm0 4a1 1 0 1 0 0-2 1 1 0 0 0 0 2Z"
            fill="#000"
            fill-rule="evenodd"
          />
        </svg>
      </button>
    </div>
  </body>,
  "container": <div>
    <button
      class="sc-bcXHqe sc-gswNZR ePFmvY jAVsth"
    >
      <svg
        fill="none"
        height="10"
        width="14"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          clip-rule="evenodd"
          d="M4 9a.5.5 0 0 1 .5-.5h9a.5.5 0 0 1 0 1h-9A.5.5 0 0 1 4 9Zm0-4a.5.5 0 0 1 .5-.5h9a.5.5 0 0 1 0 1h-9A.5.5 0 0 1 4 5Zm0-4a.5.5 0 0 1 .5-.5h9a.5.5 0 0 1 0 1h-9A.5.5 0 0 1 4 1ZM1 2a1 1 0 1 0 0-2 1 1 0 0 0 0 2Zm0 4a1 1 0 1 0 0-2 1 1 0 0 0 0 2Zm0 4a1 1 0 1 0 0-2 1 1 0 0 0 0 2Z"
          fill="#000"
          fill-rule="evenodd"
        />
      </svg>
    </button>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`testing button component testing default button 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <button
        class="sc-bcXHqe eftUQR"
      >
        click
      </button>
    </div>
  </body>,
  "container": <div>
    <button
      class="sc-bcXHqe eftUQR"
    >
      click
    </button>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`testing button component testing icon button 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <button
        class="sc-bcXHqe ePFmvY"
      >
        <svg
          fill="none"
          height="10"
          width="14"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            clip-rule="evenodd"
            d="M4 9a.5.5 0 0 1 .5-.5h9a.5.5 0 0 1 0 1h-9A.5.5 0 0 1 4 9Zm0-4a.5.5 0 0 1 .5-.5h9a.5.5 0 0 1 0 1h-9A.5.5 0 0 1 4 5Zm0-4a.5.5 0 0 1 .5-.5h9a.5.5 0 0 1 0 1h-9A.5.5 0 0 1 4 1ZM1 2a1 1 0 1 0 0-2 1 1 0 0 0 0 2Zm0 4a1 1 0 1 0 0-2 1 1 0 0 0 0 2Zm0 4a1 1 0 1 0 0-2 1 1 0 0 0 0 2Z"
            fill="#000"
            fill-rule="evenodd"
          />
        </svg>
      </button>
    </div>
  </body>,
  "container": <div>
    <button
      class="sc-bcXHqe ePFmvY"
    >
      <svg
        fill="none"
        height="10"
        width="14"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          clip-rule="evenodd"
          d="M4 9a.5.5 0 0 1 .5-.5h9a.5.5 0 0 1 0 1h-9A.5.5 0 0 1 4 9Zm0-4a.5.5 0 0 1 .5-.5h9a.5.5 0 0 1 0 1h-9A.5.5 0 0 1 4 5Zm0-4a.5.5 0 0 1 .5-.5h9a.5.5 0 0 1 0 1h-9A.5.5 0 0 1 4 1ZM1 2a1 1 0 1 0 0-2 1 1 0 0 0 0 2Zm0 4a1 1 0 1 0 0-2 1 1 0 0 0 0 2Zm0 4a1 1 0 1 0 0-2 1 1 0 0 0 0 2Z"
          fill="#000"
          fill-rule="evenodd"
        />
      </svg>
    </button>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
